viewing_party/party.py

```diff
@@ -1,23 +1,146 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie_dict = {"title": title, "genre": genre, "rating": rating}
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    '''
+    Takes two arguments, user_data and movie, where user_data 
+    is a dictionary that stores all of the movies the user has watched 
+    and movie is a dictionary which contains the information for a movie
+    
+    Adds the value of movie to the "watched" list inside of user_data
+
+    Returns user_data
+    '''
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    '''
+    adds movie to the watchlist list inside of user_data,
+    where movie is a dictionary and user_data is a dictionary with a list of dictionaries 
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    '''
+    moves title from a user's watchlist to their watched list
+    '''
+    # watchlist_list = user_data["watchlist"]
+    # print(watchlist_list)
+    for index in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][index]["title"] == title:
+            user_data["watched"].append(user_data["watchlist"].pop(index))
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    '''
+    calculates the average rating of the movies in the user's watched list
+    '''
+    avg_rating = 0.0
+    rating_sum = 0.0
+    if len(user_data["watched"]) == 0:
+        return avg_rating
+    else:
+        for movie in user_data["watched"]:
+            rating_sum += movie["rating"]
+    avg_rating = rating_sum / len(user_data["watched"])
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    current_highest_genre = None
+    current_highest_genre_score = 0

```

Comment:
Very minor style nitpick: As Python does not require you declare all variables at the top, I recommend getting in the habit of declaring them as late as possible.

There's two main benefits. The first is that if there are any early returns from the function, you get to avoid the memory allocation. The other is that usually makes the variable declaration pretty close to where it gets initialized/used, making it easier to see the tie between the variable and the code initializing/using it. At the top, a reader would need to scroll all the way back up and that could be a lot of scrolling in particularly complicated code!